barrels/Semicircles/Semicircles.mc

```diff
@@ -0,0 +1,257 @@
+                        result[0] = ((coords[0] * 0x80000000l)/Math.PI).toNumber();
+                        result[1] = ((coords[1] * 0x80000000l)/Math.PI).toNumber();
+                    } else {
+                        throw new BadCoordinateFormatException("Cannot handle input type for value " + value);
+                    }
+                    break;
+            }
+            return result;
+        }
+
+        //! Computes the cosine correction for two latitudes in semicircles
+        //! @param [Toybox::Lang::Number] lat1 Latitude 1 in semicircles
+        //! @param [Toybox::Lang::Number] lat2 Latitude 2 in semicircles
+        //! @return [Toybox::Lang::Double] Cosine correction
+        private function computeCosineCorrection(lat1, lat2) {
+            var result = (lat1.toLong() + lat2.toLong()) / 2;
+            result = (result * Math.PI) / 0x80000000;
+
+            return Math.cos(result);
+        }
+
+        //! Compute a fast distance between two points
+        //! @param [Toybox::Lang::Object] pt1 Type based off the following options
+        //!    [Toybox::Position::Location] Convert the location to semicircles
+        //!    [Toybox::Lang::Array] Assumes latitude is in radians as value 0, and longitude is in radians in value 1
+        //! @param [Toybox::Lang::Object] pt2 Type based off the following options
+        //!    [Toybox::Position::Location] Convert the location to semicircles
+        //!    [Toybox::Lang::Array] Assumes latitude is in radians as value 0, and longitude is in radians in value 1
+        public function computeDistanceInMeters(pt) {
+            var pt1 = [_lat, _lon];
+            var pt2 = convertToSemicircle(pt);
+
+            // Correct for the curvature of the earth by finding the cosine of the
+            // midpoint of the two latitudes
+            var cosineCorrection = computeCosineCorrection(pt1[0], pt2[0]);
+
+            // Compute dLat and dLon
+            var distLat = (pt1[0] - pt2[0]).toLong();
+            var distLon = ((pt1[1] - pt2[1]) * cosineCorrection).toLong();
+
+            // Compute the distance
+            return Math.sqrt((distLat * distLat) + (distLon * distLon)) / SC_PER_M;
+        }
+
+        //! Output the coordinate in radians
+        //! @return [Toybox::Lang::Array] Array of [latitude, longitude] in radians
+        public function toRadians() {
+            var result = [(_lat.toDouble() * Math.PI) / 0x80000000,
+            (_lon.toDouble() * Math.PI) / 0x80000000];
+            return result;
+        }
+
+        //! Output the coordinate in semicircles
+        //! @return [Toybox::Lang::Array] Array of [latitude, longitude] in semicircles
+        public function toSemicircles() {
+            return [_lat, _lon];
+        }
+    }
+
+
+    // Wrapping all test cases within a module will allow the compiler
+    // to eliminate the entire module when not building unit tests.
+    (:test)
+    module Test {
```

Comment:
I don't know if we have a defined standard for defining tests. Should tests be defined in a separate file to keep implementation and test code separate?

Reply:
In the updated change I moved it to a separate file.